src/main/java/com/ft/wordpressarticlemapper/resources/WordPressArticleMapperResource.java

```diff
@@ -42,16 +44,37 @@ public WordPressArticleMapperResource(BodyProcessingFieldTransformer bodyProcess
     @Path("/map")
     @Produces(MediaType.APPLICATION_JSON + CHARSET_UTF_8)
     public final WordPressContent map(NativeWordPressContent nativeWordPressContent, @Context HttpHeaders httpHeaders) {
-        Post postDetails = nativeWordPressContent.getPost();
-        String transactionId = TransactionIdUtils.getTransactionIdOrDie(httpHeaders, postDetails.getUuid(), "Publish request");
-        return getWordPressContent(nativeWordPressContent, transactionId);
+        try {
+            Post post = nativeWordPressContent.getPost();
+            if (post == null) {
+                throw new IllegalArgumentException("No post supplied");
```

Comment:
I upgraded jax-rs-transaction-id-handling library to version 2.6.1 and moved transaction id related logic as first step